doc.go

```diff
@@ -0,0 +1,56 @@
+// A Redis-backed rate limiter, based on the leaky-bucket algorithm, implemented
+// using a purely EVAL-based solution.
+//
+//	package main
+//
+//	import (
+//		"context"
+//		"net/http"
+//		"strconv"
+//		"time"
+//
+//		"github.com/go-redis/redis/v8"
+//		limiter "github.com/plsmphnx/go-redis-bucket"
+//	)
+//
+//	func main() {
+//		// Create a Redis client with appropriate configuration and error handling.
+//		r := redis.NewClient(&redis.Options{})
+//
+//		// Create a limiter that restricts calls to 10-20 per minute.
+//		l, err := limiter.NewLimiter(limiter.Config{
```

Comment:
nit: `limiter.NewLimiter` stutters. standard practice would be to just do `limiter.New()`